### PR TITLE
Fix https://github.com/Homebrew/homebrew-cask/issues/87045

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

See https://github.com/Homebrew/homebrew-cask/issues/87045 and https://github.com/Homebrew/brew/pull/8214.

### CODE REVIEW

File: Library/Homebrew/env_config.rb [hunk 1]
```diff
@@ -315,5 +318,22 @@ def make_jobs
           .call
           .to_s
     end
+
+    def cask_opts
+      Shellwords.shellsplit(ENV.fetch("HOMEBREW_CASK_OPTS", ""))
+    end
+
+    def cask_opts_quarantine
```
Comment (MikeMcQuaid) [inline Library/Homebrew/env_config.rb:326]:
```suggestion
    def cask_opts_quarantine?
```

File: Library/Homebrew/env_config.rb [hunk 2]
```diff
@@ -315,5 +318,22 @@ def make_jobs
...
+      cask_opts.reverse_each do |opt|
+        return true if opt == "--quarantine"
+        return false if opt == "--no-quarantine"
+      end
+
+      true
+    end
+
+    def cask_opts_require_sha
```
Comment (MikeMcQuaid) [inline Library/Homebrew/env_config.rb:335]:
```suggestion
    def cask_opts_require_sha?
```

**Comment by reitermarkus** [comment]
Removed the changes for `reinstall` and `uninstall` from https://github.com/Homebrew/brew/pull/8214 for now.

These should follow pretty much the same pattern.

I can have a look at those later, unless @whoiswillma beats me to it.

**Comment by whoiswillma** [comment]
I see what to do now. I'll continue to work from my branch

**Comment by whoiswillma** [comment]
Quick question: aren't there now two places where args are parsed? In `Cask::Cmd::process_options` and also in `env_config.rb`? I'm fine to continue this but it feels like a code smell

**Comment by reitermarkus** [comment]
I think once `cask` commands are using `CLI::Parser`, the parsing code duplication will be gone.

**Pull request merged**